lib/UR/Object/Type/AccessorWriter.pm

```diff
@@ -402,145 +402,142 @@ sub _resolve_bridge_logic_for_indirect_property {
-                if (@via2_join_properties > 1) {
-                    Carp::carp("via2 join not implemented :(");
-                    return;
-                }
-                my($my_property_name,$their_property_name) = @{ $via2_join_properties[0] };
-                my $crosser_template = UR::BoolExpr::Template->resolve($final_class_name, "$their_property_name in");
-
-                my $result_property_name = $to_property_meta->to;
-
-                $bridge_crosser = sub {
-                    my $bridges = shift;
-                    my @linking_values = map { $_->$my_property_name } @$bridges;
-                    my $bx = $crosser_template->get_rule_for_values(\@linking_values);
-                    my @result_objects = (@_ ? $final_class_name->get($bx->params_list, @_) : $final_class_name->get($bx) );
-
-                    if ($bridge_meta_params{'-order'} || $bridge_meta_params{'-order_by'}) {
-                        my $results_sorter = $make_results_sorter->(
-                                                $bridges,
-                                                sub { $_->$my_property_name },
-                                                sub { $_->$their_property_name } );
-                        @result_objects = $results_sorter->(\@result_objects);
-                    }
-                    return map { $_->$result_property_name } @result_objects;
-                };
-            }
+    my $rank = 0;
+    my %bridge_rankings = map  { $bridge_linker->() => $rank++ } @$bridges;
+    return sub {
+        my $results = shift;
+
+        return map { $_->[1] }
+               sort { $bridge_rankings{ $a->[0] } <=> $bridge_rankings{ $b->[0] } }
+               map { [ $results_linker->(), $_ ] }
+               @$results;
+    };
+}
 
-        } elsif ($to_property_meta->id_by and $to_property_meta->id_class_by) {
+sub _resolve_bridge_crosser_for_doubly_delegated_property {
+    my($to_property_meta, $bridge_meta_params) = @_;
+    # This property's value is doubly delegated.  The simple thing to
+    # do is to collect the bridge objects, then call the second
+    # delegation method on each bridge in a loop to collect the final
+    # results, which may trigger one query per result.  Depending on
+    # the type of delegation, the final results can be collected with
+    # one query
+
+    my($result_class_resolver, $bridge_linking_properties, $final_result_property_name, $result_filtering_property);
+    if ($to_property_meta->via) {
+        # bridges through another via-to property
+        my $second_via_property_meta = $to_property_meta->via_property_meta;
+        my $final_class_name = $second_via_property_meta->data_type;
+        if ($final_class_name and $final_class_name ne 'UR::Value' and $final_class_name->isa('UR::Object')) {
+            if ( 1 == (my @via2_join_properties = $second_via_property_meta->get_property_name_pairs_for_join)) {
```

Comment:
I did it this way to keep ```@via2_join_properties``` scoped only to the if block.  Then the ```1 ==``` went on the front so it stands out to indicate it's checking the list length